test/test_composite.jl

```diff
@@ -21,11 +21,13 @@ ftol = 1f-6
 @testset "judiVStack Unit Tests with $(nsrc) sources" for nsrc=[1, 2]
 
     # set up judiVector fr,om array
+    n = (10, 10)   # (x,y,z) or (x,z)
     dsize = (nsrc*nrec*ns, 1)
     rec_geometry = example_rec_geometry(nsrc=nsrc, nrec=nrec)
     data = randn(Float32, ns, nrec)
     d_obs = judiVector(rec_geometry, data)
-    w0 = judiWeights(rand(10, 10); nsrc=nsrc)
+    w0 = judiWeights(randn(Float32,n);nsrc=nsrc)
+    w0.weights=[randn(Float32,n) for i = 1:nsrc]
```

Comment:
?

Reply:
judiWeights(rand(10, 10); nsrc=nsrc) This will generate 2 sources with exactly the same weights. So I am thinking of giving them different random weights here for a general case.

Reply:
`w0 = judiWeights([randn(Float32,n) for i = 1:nsrc];nsrc=nsrc)`

I know it sounds nitpicky, but it's good practice to try to avoid unnecessary memory allocations at all time and get used to how to do it. Currently you allocate all the `randn` then allocate new ones and replace it which is wastefull.

Reply:
Also, w0 = judiWeights(rand(10, 10); nsrc=nsrc) will output w0 as a Float32 judiWeights but the weights are actually in Float64

Reply:
`w0 = judiWeights([randn(Float32,n) for i = 1:nsrc];nsrc=nsrc)`
This does not work because judiWeights can't tell it's too sources then the weights will be messed up.
I updated this in the latest commit by

`    weights = Array{Array}(undef, nsrc)
    for j=1:nsrc
        weights[j] = randn(Float32, n)
    end
    w0 = judiWeights(weights)
`

Reply:
That should not work because it does not take cell array.
`w0 = judiWeights(convertToCell([randn(Float32,n) for i = 1:nsrc]))`
This solves the problem and it's in the latest commit

Reply:
>That should not work because it does not take cell array.
Well that's the point that's what should be fixed, 
>This solves the problem and it's in the latest commit
This does not solve anything, it will still error if some tries it it just hides the problem.

Reply:
`function judiWeights(weights::Union{Array{Any,1},Array{Array{iDT, N},1}}; vDT::DataType=Float32) where {iDT, N}`

Reply:
Updated in the latest commit for 3 ways of input of judiWeights

Array{Array,1}
Array{Any,1}
Array{Array{Float32,2},1}

And how do they convert to the correct data types respectively